plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSink.java

```diff
@@ -129,409 +39,57 @@ public DeltaLakePageSink(
-                .setMaxPageSize(getParquetWriterPageSize(session))
-                .build();
-        CompressionCodecName compressionCodecName = getCompressionCodec(session).getParquetCompressionCodec();
-
-        try {
-            Closeable rollbackAction = () -> fileSystem.deleteFile(path);
-
-            List<Type> parquetTypes = dataColumnTypes.stream()
-                    .map(type -> {
-                        if (type instanceof TimestampWithTimeZoneType) {
-                            verify(((TimestampWithTimeZoneType) type).getPrecision() == 3, "Unsupported type: %s", type);
-                            return TIMESTAMP_MILLIS;
-                        }
-                        return type;
-                    })
-                    .collect(toImmutableList());
-
-            // we use identity column mapping; input page already contains only data columns per
-            // DataLagePageSink.getDataPage()
-            int[] identityMapping = new int[dataColumnTypes.size()];
-            for (int i = 0; i < identityMapping.length; ++i) {
-                identityMapping[i] = i;
-            }
-
-            ParquetSchemaConverter schemaConverter = new ParquetSchemaConverter(parquetTypes, dataColumnNames, false, false);
-            return new ParquetFileWriter(
-                    fileSystem.newOutputFile(path),
-                    rollbackAction,
-                    parquetTypes,
-                    dataColumnNames,
-                    schemaConverter.getMessageType(),
-                    schemaConverter.getPrimitiveTypes(),
-                    parquetWriterOptions,
-                    identityMapping,
-                    compressionCodecName,
-                    trinoVersion,
-                    false,
-                    Optional.empty(),
-                    Optional.empty());
-        }
-        catch (IOException e) {
-            throw new TrinoException(DELTA_LAKE_BAD_WRITE, "Error creating Parquet file", e);
-        }
-    }
-
-    private FileWriter createRecordFileWriter(Path path)
-    {
-        Properties schema = buildHiveSchema(dataColumnNames, dataColumnTypes);
-        return new RecordFileWriter(
-                path,
-                dataColumnNames,
-                fromHiveStorageFormat(PARQUET),
-                schema,
-                PARQUET.getEstimatedWriterMemoryUsage(),
-                conf,
-                typeManager,
-                DateTimeZone.UTC,
-                session);
-    }
-
-    private Page getDataPage(Page page)
-    {
-        Block[] blocks = new Block[dataColumnInputIndex.length];
-        for (int i = 0; i < dataColumnInputIndex.length; i++) {
-            int dataColumn = dataColumnInputIndex[i];
-            blocks[i] = page.getBlock(dataColumn);
-        }
-        return new Page(page.getPositionCount(), blocks);
-    }
-
-    private static Page extractColumns(Page page, int[] columns)
+    protected void collectDataFileInfo(DataFileInfo dataFileInfo)
```

Comment:
Can we unify this and the CDF version and put it in the base class? Looks like the only difference is that this one converts to Slice immediately, while the CDF version collects DataFileInfos and then converts to Slices at the end,

Reply:
ok I will take a look